src/section_handlers/document_tracking.py

```diff
@@ -0,0 +1,154 @@
+        """
+        Checks whether all version numbers in /document/tracking/revision_history match semantic versioning.
+        semantic version is defined in version_t definition
+        see: https://docs.oasis-open.org/csaf/csaf/v2.0/csd01/csaf-v2.0-csd01.html#3111-version-type and
+        section 9.1.5 Conformance Clause 5: CVRF CSAF converter
+        """
+
+        pattern = (
+            r'^((0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)'
+            r'(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?)$'
+        )
+        return all(re.match(pattern, revision['number']) for revision in revision_history)
+
+    @staticmethod
+    def _as_int_tuple(text: str) -> Tuple[int]:
+        """ Converts string of dotted numbers into tuple of ints """
+        try:
+            return tuple(int(part) for part in text.split('.'))
+        except ValueError:
+            return (sys.maxsize, )
+
+    def _add_current_revision_to_history(self, root_element, revision_history) -> None:
+        """
+        If the current version is missing in Revision history, and the Current revision is X.Y.Z, and the last item
+        in history is X.(Y-1).Z, add the current version to the history
+
+        Throws ValueError when didn't match these conditions
+        """
+        try:
+            current_minor_version = self._as_int_tuple(root_element.Version.text)[1]
+
+            latest_history_revision = revision_history[-1]['number']
+            latest_history_minor_version = self._as_int_tuple(latest_history_revision)[1]
+        except (IndexError, AttributeError):
+            raise ValueError('Missing minor version number in the current or latest history Version')
+
+        if current_minor_version - latest_history_minor_version == 1:
+            revision_history.append(
+                {
+                    'date': root_element.CurrentReleaseDate.text,
+                    'number': root_element.Version.text,
+                    'summary': "Added by CVRF-CSAF-Converter",
+                    # Extra vars
+                    'number_cvrf': root_element.Version.text,
+                    'version_as_int_tuple': self._as_int_tuple(root_element.Version.text),
+                }
+            )
+
+        else:
+            raise ValueError('Can not handle such version difference between The current and the history"s last version')
+
+
+    def _reindex_history_to_integers(self, root_element, revision_history):
+        logging.info('Some version numbers in /document/tracking/revision_history do not match semantic versioning. '
+                     'Reindexing to integers.')
+
+        revision_history_sorted = sorted(revision_history, key=itemgetter('version_as_int_tuple'))
+
+        for rev_number, revision in enumerate(revision_history_sorted, start=1):
+            revision['number'] = rev_number  # Changing the type from str to int
```

Comment:
I think there is a minor misunderstanding here: the number is always a string (according to the CSAF JSON schema) - even when it is called integer versioning...

```suggestion
            revision['number'] = str(rev_number)